salesagility/SuiteCRM
# Add oauth external providers module

## Description




- Add module to configure external OAuth providers
- Several email related fixes

## Motivation and Context

Ideally the external OAuth provider configuration should be configured through the ui interface.

## How To Test This



1. Configure Microsoft connection provider using a configuration similar to the following
![external_outh_provider_config_example](https://user-images.githubusercontent.com/1140849/207900253-e2216831-50c6-456d-a3dc-7e2705233f53.png)


2. Go to Profile > External OAuth connection module and create new personal connection

3. Select Microsoft > Click Authenticate > Popup should appear

4. Add your credentials > Authorize > Connection fields should be field in with token information

5. Add a name > Save

6. Go to InboundEmail record

7. Change Auth Type from basic to OAuth

8. You should be able to see the records you and the ones you have access to by security groups

9 . Select a record > Save > The record should correctly save

2. Go to Profile > External OAuth connection module and create new personal connection

3. Select Microsoft > Click Authenticate > Popup should appear

4. Add your credentials > Authorize > Connection fields should be field in with token information

5. Add a name > Save

6. Go to InboundEmail record

7. Change Auth Type from basic to OAuth

8. You should be able to see the records you and the ones you have access to by security groups

9 . Select a record > Save > The record should correctly save

10. Go to emails module. You should be able to see the emails sorted by received date

11. Try the filter panel you should be able to filter

12. You should not be able to sort or set sorting direction

13. You should be able to mark emails as read and unread

14. You should be able to flag and un-flag emails

15. You should be able to delete emails

16. You should be able to import emails

17. you should be able to go to the email detail view page

18. You should be able to see the email information in the detail view page

19 Configure  a group inbound emails. You will need to create a group ExternalOauthConnection

20. Auto email import for cases should work for group emails

21. Inbound email repair should work

## Types of changes


### Final checklist
